extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -125,33 +149,46 @@ class LifecycleTask extends BackbeatTask {
     _getMPUs(bucketData, bucketLCRules, params, log, done) {
         const req = this.s3target.listMultipartUploads(params);
         attachReqUids(req, log);
-        return req.send((err, data) => {
+        async.waterfall([
+            next => req.send((err, data) => {
+                if (err) {
+                    log.error('error checking buckets MPUs', {
+                        method: 'LifecycleTask.processBucketEntry',
+                        error: err,
+                    });
+                    return next(err);
+                }
+                return next(null, data);
+            }),
+            (data, next) => {
+                if (data.IsTruncated) {
+                    // re-queue to kafka with `NextUploadIdMarker` &
+                    // `NextKeyMarker`
+                    const entry = Object.assign({}, bucketData, {
+                        details: {
+                            keyMarker: data.NextKeyMarker,
+                            uploadIdMarker: data.NextUploadIdMarker,
+                        },
+                    });
+                    return this.sendBucketEntry(entry, err => {
+                        if (!err) {
+                            log.debug(
+                                'sent kafka entry for bucket consumption', {
+                                    method: 'LifecycleTask._getMPUs',
+                                });
+                            this._publishedNextListingTask = true;
+                        }
+                        return next(null, data);
+                    });
+                }
+                return process.nextTick(() => next(null, data));
+            },
+        ], (err, data) => {
             if (err) {
-                log.error('error checking buckets MPUs', {
-                    method: 'LifecycleTask.processBucketEntry',
-                    error: err,
-                });
                 return done(err);
             }
-            if (data.IsTruncated) {
-                // re-queue to kafka with `NextUploadIdMarker` & `NextKeyMarker`
-                const entry = Object.assign({}, bucketData, {
-                    details: {
-                        keyMarker: data.NextKeyMarker,
-                        uploadIdMarker: data.NextUploadIdMarker,
-                    },
-                });
-                this.sendBucketEntry(entry, err => {
-                    if (!err) {
-                        log.debug('sent kafka entry for bucket consumption', {
-                            method: 'LifecycleTask._getMPUs',
-                        });
-                        this._publishedNextListingTask = true;
-                    }
-                });
-            }
             this._compareMPUUploads(bucketData, bucketLCRules,
-                data.Uploads, log);
+                                    data.Uploads, log);
```

Comment:
trivial, but I think this indentation style is really detrimental to reading the code. it's not worth changing rn, but maybe we should decide on a standard to improve readability/consistency for future backbeat contributors :)

Reply:
This was an involuntary change that I should not have committed (my beloved editor e\*\*\*s indents this way when pressing TAB), though personally speaking I tend to prefer this style, but point taken, I'm down for one or the other depending on which one everyone is most comfortable with :wink: